### ISSUE TITLE
Replace the top menu with a side menu

### ISSUE THREAD

**Issue by IgorA100** [ZoneMinder/zoneminder#4077]
I propose to discuss the option of adding a side menu like here: https://www.bootstrap.gallery/demos/milestone-admin-dashboard/index.html#

Sources: https://www.bootstrap.gallery/details/milestone---bootstrap-admin-panel

**Comment by connortechnology** [comment]
Go for it, but it has to be configurable. Not everyone wants that.

**Comment by baudneo** [comment]
Seems like a good idea, I do agree on configurable.

**Comment by IgorA100** [comment]
Those. I have to add an option in settings to be able to use the current menu or a new one.
Right ?

**Comment by baudneo** [comment]
> Those. I have to add an option in settings to be able to use the current menu or a new one. Right ?

yes

**Comment by connortechnology** [comment]
I may actually get around to merging some code that would at least provide a sidebar if turned on.  I want to also make the top navbar really completely configurable in terms of content.

Anyways, going to re-open this and put it on the 1.39 roadmap.

**Comment by IgorA100** [comment]
I wanted to implement this menu, but I haven't managed to structure the sections and levels of the menu in my head yet, and also understand what to display in the side menu and what in the top one.
Later I worked on the Montage page for viewing recorded events with the new Timeline and haven't finished it yet (there's really a lot of work and nuances...). Therefore, the question with the side menu remains open.
Perhaps it's worth discussing what sections and levels are needed in the menu?

**Comment by IgorA100** [comment]
@connortechnology 

Here's an interesting option: https://github.com/azouaoui-med/pro-sidebar-template/tree/master
Demo: https://azouaoui-med.github.io/pro-sidebar-template
It seems to work fine on mobile devices too.
Take a look when you have time and tell me what you think.

**Comment by connortechnology** [comment]
Looks fine.  No problem with it.  Side bars are not hard these days, lots of options.

Point Is, let's add it, but it has to be configurable.  Some people will want top menus as is, others may want a left menu.

**Comment by IgorA100** [comment]
> Some people will want top menus as is, others may want a left menu.

That is, duplicate the information in the left and top menu?
And make it possible to switch between using the top/side menu

> Side bars are not hard these days, lots of options.

Yes, but many of them are either not very convenient, or buggy, or have complicated code....

**Comment by connortechnology** [comment]
Yes.  I want to make the content in the header configurable as well.  But that's for 1.39+

**Comment by IgorA100** [comment]
Ok, let's discuss.

**Comment by IgorA100** [comment]
@connortechnology 
Just a mockup of the sidebar layout.
Isaac, what are your thoughts

![Image](https://github.com/user-attachments/assets/360ed819-69d5-452c-b738-de970ab57ce6)
![Image](https://github.com/user-attachments/assets/088a3892-ad20-43ed-a37d-9470d9b73742)

**Comment by IgorA100** [comment]
@connortechnology 
Menu is already filled. The layout is partially corrected.
Isaac, I need your opinion. What do you like and what do you not like.

![Image](https://github.com/user-attachments/assets/bd9efac7-d4b9-4613-8272-48884096e172)
![Image](https://github.com/user-attachments/assets/1a8dd0fd-8c11-49f6-b246-1200bfad89a3)

**Comment by IgorA100** [comment]
Mobile view
![Image](https://github.com/user-attachments/assets/849be418-16b7-4302-9723-ee4e333e33a5)
![Image](https://github.com/user-attachments/assets/c70b09d6-4a94-4a44-8dff-1660b953d2ad)

**Comment by IgorA100** [comment]
I liked it in one row and with scrolling. It's convenient on a mobile device!

**Comment by IgorA100** [comment]
Made the scroll bar prettier and increased the height of the status bar.
The scroll bar is specially made thin so as not to take up too much space and not to irritate. No one will cling to the scroll bar on a mobile device anyway, and the shift will be a grab on the status bar itself.
I like it...

![Image](https://github.com/user-attachments/assets/337db1c6-7f7a-457c-a036-548253d0f99b)

**Comment by IgorA100** [comment]
I decided to move the filters to the Side bar.
This is what I got:

There is still a lot of work to do...
I am waiting for criticism

![Image](https://github.com/user-attachments/assets/daeab8b0-a8c8-47f1-8db0-a6fa4af3143f)

![Image](https://github.com/user-attachments/assets/5f9685e7-3959-47b4-8d30-482c3b22fd00)

**Comment by baudneo** [comment]
I am loving it

**Comment by IgorA100** [comment]
So, the work on the new menu is almost finished.
For touch screens, Swipe from left to right is available to display the menu.

The active menu item is highlighted.
The menu item for setting the current filter is always located at the top and does not move when scrolling other menu items. This menu item for displaying the current filter is currently implemented only for the pages: 'console', 'montage', 'montagereview', 'watch' and 'events'. Maybe for some other pages the filter should be moved to a pull-out block?

Please give your opinion on the menu items. The order of the items has been slightly changed and the Events page item has been added.

![Image](https://github.com/user-attachments/assets/7fabc036-9652-42a4-9db2-2bee98376779)
![Image](https://github.com/user-attachments/assets/7633133c-b5ef-425e-ad8d-30107433adc5)
![Image](https://github.com/user-attachments/assets/be78afc8-8e14-4b5a-a64c-3cd8857ffb8e)
![Image](https://github.com/user-attachments/assets/d0f7a40c-5647-4192-9e13-49007eb80e11)
